src/data/list/basic.lean

```diff
@@ -1584,6 +1584,25 @@ begin
     exact lt_of_add_lt_add_left' (lt_of_le_of_lt h $ add_lt_add_right (lt_of_not_ge h') _) }
 end
 
+/-- A list of natural numbers with positive sum must have positive length. -/
+-- This could perhaps be generalised, to ordered additive monoids bounded below by zero.
+-- I'm not sure what those are called.
```

Comment:
You don't need the bounded below... as soon as the sum is nonzero... bingo

Reply:
Good call. I will fix this.

Reply:
In fact, I'll move it to the other PR, with facts about sums in an arbitrary monoid.